File: leetcode/src/scala/MaximumSubarrayI.scala
object MaximumSubarrayI {
    def findMaxAverage(nums: Array[Int], k: Int): Double = {
        var maxSum = 0.0
        var currentSum = 0.0
        Array.range(0, nums.length).foreach(i => {
            if (i < k) {
                currentSum += nums(i)
                maxSum = currentSum
            } else {
                currentSum -= nums(i - k)
                currentSum += nums(i)
                if (currentSum > maxSum) maxSum = currentSum
            }
        })
        maxSum / k
    }
}

File: leetcode/src/python/maximum_subarray_i.py
from typing import List

class MaximumSubarrayI:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_sum = 0
        current_sum = 0.0
        for i, num in enumerate(nums):
            if i < k:
                current_sum += num
                max_sum = current_sum
            else:
                current_sum -= nums[i - k]
                current_sum += num
                if current_sum > max_sum:
                    max_sum = current_sum
        return max_sum / k